p2p/src/peer_manager/mod.rs

```diff
@@ -986,6 +986,17 @@ where
             PeerManagerEvent::RemoveReserved(address) => {
                 self.peerdb.remove_reserved_node(address);
             }
+            PeerManagerEvent::ListBanned(response) => {
+                response.send(self.peerdb.list_banned().cloned().collect())
+            }
+            PeerManagerEvent::Ban(address, response) => {
+                self.peerdb.ban(address);
+                response.send(Ok(()));
+            }
+            PeerManagerEvent::Unban(address, response) => {
+                self.peerdb.unban(&address);
+                response.send(Ok(()));
+            }
```

Comment:
Would it be useful for the response to include whether ban/unban has been successful?

Reply:
I think they both can't fail. Ban will always succeed because we just store the IP address in the DB. Unban will always succeed because the specified IP won't be banned after the operation.

BTW, it just occurred to me that we should also disconnect selected peers when we ban them. I will do that in the next PR.

Reply:
Yea what I meant is something like:

* When you ban an address that's already banned, report that
* When you unban an address that has not been banned.

However, usefulness of that is questionable, so I don't mind not having it.